docs/roleassumption.md

```diff
@@ -0,0 +1,414 @@
+# Creating clusters using cross account role assumption using KIAM
+
+This document outlines the list of steps to create the target cluster via cross account role assumption using [KIAM]https://github.com/uswitch/kiam. 
+KIAM lets the controller pod(s) to assume an AWS role that enables them create AWS resources necessary to create an
+operational cluster. This way we wouldn't have to mount any AWS credentials or load environment variables to 
+supply AWS credentials to the CAPA controller. This is automatically taken care by the KIAM components.
+Note: If you dont want to use KIAM and rather want to mount the credentials as secrets, you may still achieve cross 
+account role assumption by using multiple profiles. 
+
+### Glossory
+
+* Target Account - The account where the cluster is created
+* Source Account - The AWS account where the CAPA controller runs. 
+
+## Assumptions
+1. The CAPA controllers are running in an AWS account and you want to create the target cluster in another AWS account.
```

Comment:
This is more of a result of the steps instead of a prior assumption, maybe this should be under a #goals section?